lib/icingadb/redisconnection.cpp

```diff
@@ -233,12 +254,48 @@ void RedisConnection::Connect(asio::yield_context& yc)
 	for (;;) {
 		try {
 			if (m_Path.IsEmpty()) {
-				Log(LogInformation, "IcingaDB")
-					<< "Trying to connect to Redis server (async) on host '" << m_Host << ":" << m_Port << "'";
+				if (m_TLSContext) {
+					Log(LogInformation, "IcingaDB")
+						<< "Trying to connect to Redis server (async, TLS) on host '" << m_Host << ":" << m_Port << "'";
+
+					auto conn (Shared<AsioTlsStream>::Make(m_Strand.context(), *m_TLSContext, m_Host));
+					auto& tlsConn (conn->next_layer());
+
+					if (!m_CaPath.IsEmpty()) {
+						auto native (tlsConn.native_handle());
+
+						X509_VERIFY_PARAM_set1_host(SSL_get0_param(native), m_Host.CStr(), 0);
+						SSL_set_verify(native, SSL_VERIFY_PEER, NULL);
+					}
```

Comment:
Do we really only want to do this if a CA is explicitly specified? I'd hope OpenSSL would use the system trust store by default (would have to check that, possibly `AsioTlsStream` changes this) which should be a fine default.

@lippserd Any opinions on this? Given that there are also discussions on how to handle TLS elsewhere (notably Web), I think we should decide on what new behavior we want and write all new code consistent with that.